R/derive_extreme_event.R

```diff
@@ -156,37 +286,62 @@
 #'   mode = "first",
 #'   set_values_to = exprs(
 #'     PARAMCD = "WSP",
-#'     PARAM = "Worst Sleeping Problems"
-#'   )
-#' )
-#'
-#' # Use different mode by event
+#'     PARAM = "Worst Sleeping Problem"
+#'   ),
+#'   keep_source_vars = exprs(everything())
+#' ) %>%
+#' select(-STUDYID)
+#'
+#' @caption Specifying different arguments across `event()` objects
+#' @info Here we consider a Hy's Law use case. We are interested in
+#'   knowing whether a subject's Alkaline Phosphatase has ever been
+#'   above twice the upper limit of normal range. If so, i.e. if
+#'   `CRIT1FL` is `Y`, we are interested in the record for the first
+#'   time this occurs, and if not, we wish to retain the last record.
+#'   As such, for this case now we need to vary our usage of the
+#'   `mode` argument dependent on the `event()`.
+#'
+#'  - In first `event()`, since we simply seek the first time that
+#'    `CRIT1FL` is not `NA`, it's enough to specify the `condition`,
+#'    because we inherit `order` and `mode` from the main
+#'    `derive_extreme_event()` call here which will automatically
+#'    select the first occurrence by `AVISITN`.
+#'  - In the second `event()`, we select the last record among the
+#'    full set of records where `CRIT1FL` are all `NA` by additionally
+#'    specifying `mode = "last"` within the `event()`.
+#'  - Note now the usage of `keep_source_vars = exprs(AVISITN)`
+#'    rather than `everything()` as in the previous example. This
+#'    is done to ensure `CRIT1` and `CRIT1FL` are not populated for
+#'    the new records.
+#'
+#' @code
 #' adhy <- tribble(
-#'   ~USUBJID, ~AVISITN, ~CRIT1FL,
-#'   "1",             1, "Y",
-#'   "1",             2, "Y",
-#'   "2",             1, "Y",
-#'   "2",             2, NA_character_,
-#'   "2",             3, "Y",
-#'   "2",             4, NA_character_
+#'   ~USUBJID, ~AVISITN,              ~CRIT1,      ~CRIT1FL,
+#'   "1",             1, "ALT > 2 times ULN", NA_character_,
+#'   "1",             2, "ALT > 2 times ULN", NA_character_,
+#'   "2",             1, "ALT > 2 times ULN", NA_character_,
+#'   "2",             2, "ALT > 2 times ULN", "Y",
+#'   "2",             3, "ALT > 2 times ULN", NA_character_,
+#'   "2",             4, "ALT > 2 times ULN", "Y"
```

Comment:
This is not ADaM compliant. Either `CRIT1` should be `NA` if `CRIT1FL` is `NA` or `CRIT1FL` should be `"N"` if the criterion is not fulfilled.

Reply:
good point updated thanks